src/ZkFold/Symbolic/Compiler/ArithmeticCircuit/Internal.hs

```diff
@@ -128,19 +156,29 @@ instance
 
 instance
   ( Arithmetic a, Binary a, Representable i, Binary (Rep i), Ord (Rep i)
-  , o ~ U1) => MonadCircuit (Var i) a (State (ArithmeticCircuit a i o)) where
-
-    unconstrained witness = do
-      let v = toVar @a witness
+  , o ~ U1) => MonadCircuit (Var a i) a (State (ArithmeticCircuit a i o)) where
+
+    unconstrained witness' = do
+      let witness :: Witness (SysVar i) a
```

Comment:
`witness` definition can be inlined to its only usage to simplify implementation